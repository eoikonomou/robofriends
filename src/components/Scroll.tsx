import * as React from 'react';

type Props = {
    children?: JSX.Element;
}
const Scroll = (props: Props) => {
    return (
        <div style={{ overflowY: 'scroll', border: '5px solid black', height: '790px' }}>
            {props.children}
        </div>
    );
}

export default Scroll;
