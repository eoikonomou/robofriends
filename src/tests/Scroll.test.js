import React from 'react';
import { shallow } from 'enzyme';
import Scroll from '../components/Scroll';

const mockProps = {
    children: <h1 key={'mockElement'}>Mock child</h1>
};
let wrapper;
beforeEach(() => {
    wrapper = shallow(<Scroll {...mockProps} />);
});

it('expect Scroll to render properly', () => {
    expect(wrapper).toMatchSnapshot();
});

it('expect passed children to be rendered', () => {
    expect(wrapper.contains(mockProps.children)).toBe(true);
});
