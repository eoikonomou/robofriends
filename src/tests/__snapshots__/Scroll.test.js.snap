// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect Scroll to render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll>
    <h1>
        Mock child
    </h1>
</Scroll>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Mock child
</h1>,
      "style": Object {
        "border": "5px solid black",
        "height": "790px",
        "overflowY": "scroll",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "mockElement",
      "nodeType": "host",
      "props": Object {
        "children": "Mock child",
      },
      "ref": null,
      "rendered": "Mock child",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Mock child
</h1>,
        "style": Object {
          "border": "5px solid black",
          "height": "790px",
          "overflowY": "scroll",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "mockElement",
        "nodeType": "host",
        "props": Object {
          "children": "Mock child",
        },
        "ref": null,
        "rendered": "Mock child",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
